recipes/cli11/all/conanfile.py

```diff
@@ -12,20 +13,46 @@ class CLI11Conan(ConanFile):
     license = "BSD-3-Clause"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/CLIUtils/CLI11"
-    topics = "cli-parser", "cpp11", "no-dependencies", "cli", "header-only"
-    package_type = "header-library"
+    topics = "cli-parser", "cpp11", "no-dependencies", "cli"
+    # Correct value not autodetected by Conan as this is always a static-library
+    # config_options/configure set the proper value in this case
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    options = {
+        "header_only": [True, False]
+    }
+    default_options = {
+        "header_only": False
+    }
+
     @property
     def _min_cppstd(self):
         return "11"
+    
+    @property
+    def _supports_compilation(self):
+        return Version(self.version) >= 2.3
+    
+    def config_options(self):
+        if not self._supports_compilation:
+            del self.options.header_only
+            self.package_type = "header-library"
+    
+    def configure(self):
+        if self._supports_compilation:
+            if self.options.header_only:
+                self.package_type = "header-library"
+            else:
+                self.package_type = "static-library"
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def package_id(self):
-        self.info.clear()
+        if self.info.options.get_safe("header_only", True):
```

Comment:
```suggestion
        if self.info.options.get_safe("header_only"):
```

I guess `info.options.get_safe` does not have support for defaults, only `self.options.get_safe` has it.